src/Tasks/CultureInfoCache.cs

```diff
@@ -17,907 +14,22 @@ namespace Microsoft.Build.Tasks
     /// </summary>
     internal static class CultureInfoCache
     {
-        private static readonly HashSet<string> ValidCultureNames;
-
-        static CultureInfoCache()
-        {
-            ValidCultureNames = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
-
-#if !FEATURE_CULTUREINFO_GETCULTURES
-            if (!AssemblyUtilities.CultureInfoHasGetCultures())
-            {
-                ValidCultureNames = HardcodedCultureNames;
-                return;
-            }
-#endif
-
-            foreach (CultureInfo cultureName in AssemblyUtilities.GetAllCultures())
-            {
-                ValidCultureNames.Add(cultureName.Name);
-            }
-
-            // https://docs.microsoft.com/en-gb/windows/desktop/Intl/using-pseudo-locales-for-localization-testing
-            // These pseudo-locales are available in versions of Windows from Vista and later.
-            // However, from Windows 10, version 1803, they are not returned when enumerating the
-            // installed cultures, even if the registry keys are set. Therefore, add them to the list manually.
-            var pseudoLocales = new[] { "qps-ploc", "qps-ploca", "qps-plocm", "qps-Latn-x-sh" };
-
-            foreach (string pseudoLocale in pseudoLocales)
-            {
-                ValidCultureNames.Add(pseudoLocale);
-            }
-        }
-
         /// <summary>
         /// Determine if a culture string represents a valid <see cref="CultureInfo"/> instance.
         /// </summary>
         /// <param name="name">The culture name.</param>
         /// <returns>True if the culture is determined to be valid.</returns>
         internal static bool IsValidCultureString(string name)
         {
-            return ValidCultureNames.Contains(name);
+            try
+            {
+                _ = CultureInfo.GetCultureInfo(name);
```

Comment:
We're going from a hashset lookup to try-catching a function call. This will run for every resource in a project, what's the perf impact here?

Reply:
Looks like it does a `TryGetValue` on a dictionary behind the scenes. https://source.dot.net/#System.Private.CoreLib/CultureInfo.cs,fec605b3b773ab26

Reply:
I didn't notice the discussion before, it looks like we're okay with this change so long as RPS doesn't catch this.

I also didn't realize that the previous valid cultures are calculated on each run.

Reply:
Try itself is fast as long as it doesn't hit the catch ([StackOverflow](https://stackoverflow.com/questions/1308432/do-try-catch-blocks-hurt-performance-when-exceptions-are-not-thrown)), and I imagine the cache is small enough that the dictionary lookup will be faster than looking through a list of valid cultures. My biggest worry would be if calling into the runtime causes us to load an assembly we otherwise didn't repeatedly. Not sure about that.

Reply:
> My biggest worry would be if calling into the runtime causes us to load an assembly we otherwise didn't repeatedly. Not sure about that.

The Globalization code residing inside the core library. so this change wouldn't make any difference. Or are you referring to something else?